Description: hébergement handy fotoo par défaut
 .
 handylinux-thunar-actions (0.3-7) unstable; urgency=medium
 .
   * utilisation de libreoffice pour la conversion en PDF by thuban
Author: team handylinux <dev@handylinux.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- handylinux-thunar-actions-0.3.orig/scripts/pixup
+++ handylinux-thunar-actions-0.3/scripts/pixup
@@ -4,6 +4,8 @@
 v_=$IFS
 IFS="
 "
+LIMIT=2097152  # Taille maximale en octets de l'image pouvant être envoyée sur fotoo
+
 # trads
 IMGSENT="Image sent"
 ERROR="Error. Check file name"
@@ -59,19 +61,33 @@ genhtml() {
     echo "$TEMP"
 }
 
+
 for i in $@; do
     TYPE="$(file --mime-type -b $i)"
     EXT="${i##*.}"
-    URL=$(curl -F "file=@$i" -F "format=$TYPE"\
-        -F "delete-day=0" -F "first-view=0"\
-        "https://framapic.org" | grep -Eo "https://framapic.org/(.*?).$EXT" | head -n1 | cut -d'"' -f1)
-
-    if [ $? -eq 0 -a -n "$URL" ]; then
-        LOCALHTML=$(genhtml "$URL")
-        notify-send "$IMGSENT : $i" &
-        x-www-browser "$LOCALHTML" &
+    if [ $(stat -c %s "$i") -gt $LIMIT ]; then
+        echo "Upload sur framapic"
+        # si taille supérieure à la limite de l'hébergeur handylinux, sur framapic
+        URL=$(curl -F "file=@$i" -F "format=$TYPE"\
+            -F "delete-day=0" -F "first-view=0"\
+            "https://framapic.org" | grep -Eo "https://framapic.org/(.*?).$EXT" | head -n1 | cut -d'"' -f1)
+
+        if [ $? -eq 0 -a -n "$URL" ]; then
+            LOCALHTML=$(genhtml "$URL")
+            notify-send "$IMGSENT : $i" &
+            x-www-browser "$LOCALHTML" &
+        else
+            notify-send "$ERROR" &
+        fi
     else
-        notify-send "$ERROR" &
+        echo "Upload sur l'hébergeur handylinux"
+        URL=$(curl -i -F name="$(whoami)-$i" -F private=0 -F "upload=@$i" "https://handylinux.org/fotoo/?upload" | grep '^Location' | sed 's/^Location: //')
+        if [ $? -eq 0 -a -n "$URL" ]; then
+            notify-send "$IMGSENT : $i" &
+            x-www-browser "$URL" &
+        else
+            notify-send "$ERROR" &
+        fi
     fi
 
 done
